### PR TITLE
revise sending SR notifications assuming whole hour deadlines

### PR DESCRIPTION
**Pull request by 70ray**
This is the second stage of the revised method of sending SR end notifications. It should not be merged until at least 42 days after #987 or SR periods that have not been rounded up may fail to send notifications.

Sandbox at: https://www.pgdp.org/~rp31/c.branch/improve_sr_notify

### CODE REVIEW

File: crontab/finish_smoothreading.php [hunk 1]
```diff
@@ -15,55 +15,38 @@
...
         }
     }
-}
-
-$output .= "finish_smoothreading.php executed.\n";
-
-// Don't output anything if no work was done
-if ($any_work_done) {
-    echo $output;
+    echo "finish_smoothreading.php executed.\n";
```
Comment (cpeel) [inline crontab/finish_smoothreading.php:51]: If we're removing the starting "Checking..." line we should remove this one too.

**Review by cpeel — APPROVED** [review]

File: crontab/finish_smoothreading.php (hunk 1 above)
Comment (70ray) [reply crontab/finish_smoothreading.php:51]: > If we're removing the starting "Checking..." line we should remove this one too.

I changed the "Checking..." line to "Found..." and moved it into the conditional block. So this line should also be moved. I'll do that.

Comment (cpeel) [reply crontab/finish_smoothreading.php:51]: You are correct, I missed that - thanks!

**Review by chrismiceli — APPROVED** [review]

**Review by srjfoo — APPROVED** [review]

**Review by windymilla — APPROVED** [review]

**Comment by cpeel** [comment]
Note: We need to merge this in _after_ the February deployment to PROD.

**Comment by cpeel** [comment]
Note to self: need to update dpadmin's crontab with the time change post-deploy.

**Pull request merged**